### PR TITLE
CME-26 upgrade to spring boot 3

### PR DESCRIPTION
**Pull request by rapolaskaseliscgi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CME-26

### Change description ###
Update to Spring Boot 3

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -197,11 +198,15 @@ if "$cygwin" || "$msys" ; then
...
 fi
 
-# Collect all arguments for the java command;
-#   * $DEFAULT_JVM_OPTS, $JAVA_OPTS, and $GRADLE_OPTS can contain fragments of
-#     shell script including quotes and variable substitutions, so put them in
-#     double quotes to make sure that they get re-expanded; and
-#   * put everything else in single quotes, so that it's not re-expanded.
+
+# Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
Comment (kiran-yenigala-hmcts) [inline gradlew:203]: we can add JVM options in gradle.properties file

Please see this once: https://github.com/hmcts/rd-professional-api/blob/master/gradle.properties

File: src/main/java/uk/gov/hmcts/reform/refunds/repository/RefundsRepository.java [hunk 2]
```diff
@@ -42,11 +42,11 @@ Optional<List<Refund>> findByRefundStatusAndUpdatedByIsNotAndServiceTypeInIgnore
     Optional<List<Refund>> findByNotificationSentFlag(String notificationSentFlag);
 
     @Query("select rf from Refund rf "
-        + "where rf.dateUpdated NOT between ?1 and ?2 AND (rf.refundStatus = 'Approved' or rf.refundStatus = 'Accepted')")
+        + "where rf.dateUpdated NOT between ?1 and ?2 AND (rf.refundStatus.name = 'Approved' or rf.refundStatus.name = 'Accepted')")
     List<Refund> findByDatesBetween(Date fromDate, Date toDate);
 
     @Query("select rf from Refund rf "
-        + "where rf.paymentReference = ?1  AND (rf.refundStatus = 'Approved' or rf.refundStatus = 'Accepted')"
+        + "where rf.paymentReference = ?1  AND (rf.refundStatus.name = 'Approved' or rf.refundStatus.name = 'Accepted')"
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/reform/refunds/repository/RefundsRepository.java:49]: why is this changed now?

**Review by kiran-yenigala-hmcts — COMMENTED** [review]
Minor Comments added

File: gradlew (hunk 1 above)
Comment (rapolaskaseliscgi) [reply gradlew:203]: This is auto-generated (downloaded) file while running `./gradlew wrapper --gradle-version latest` command to update gradle wrapper.

File: src/main/java/uk/gov/hmcts/reform/refunds/repository/RefundsRepository.java (hunk 2 above)
Comment (rapolaskaseliscgi) [reply src/main/java/uk/gov/hmcts/reform/refunds/repository/RefundsRepository.java:49]: Newer hibernate (>=6.2) is more strict on data types, and it couldn't compare Enum with string, therefore needed to change to compare string with string

File: src/main/java/uk/gov/hmcts/reform/refunds/config/security/SpringSecurityConfiguration.java [hunk 3]
```diff
@@ -34,195 +34,133 @@
...
-                "/swagger-resources/**",
-                "/swagger-ui/**",
-                "/v3/**",
-                "/refdata/**",
+    protected SecurityFilterChain publicFilterChain(HttpSecurity http) throws Exception {
+        http
+            .securityMatcher(
+                "/favicon.ico",
                 "/health",
                 "/health/liveness",
```
[the remark this answers is not in the document]
Comment (rapolaskaseliscgi) [reply src/main/java/uk/gov/hmcts/reform/refunds/config/security/SpringSecurityConfiguration.java:87]: I'd say we should leave it as is, and I've removed `security.anonymousPaths` property from application.yaml not to confuse. It wasn't used in this project, and extraction of those urls to properties file has nothing to do with spring boot upgrade.

**Review by kiran-yenigala-hmcts — APPROVED** [review]

**Review by Ollie-Smith — APPROVED** [review]
reviewed
